### PR TITLE
Integrate danger-simplecov_json

### PR DESCRIPTION
**Pull request by neua-wd**
## What happened
- Use `danger-simplecov_json` to report the test suite code coverage
- Addresses issue https://github.com/nimblehq/rails-templates/issues/221
 
## Proof Of Work
- Used in The 1 project 
danger code coverage

### CODE REVIEW

File: Gemfile.tt [hunk 1]
```diff
@@ -59,6 +60,7 @@ group :test do
   gem 'json_matchers' # Validate the JSON returned by your Rails JSON APIs
   gem 'webmock' # Library for stubbing and setting expectations on HTTP requests in Ruby
   gem 'simplecov', require: false # code coverage analysis tool for Ruby
```
Comment (andyduong1920) [inline Gemfile.tt:62]: Should we remove this simplecov @neua-wd 🤔 as we have the `simplecov-json` there? or the simplecov-json is a plugin for the `core simplecov`

Comment (neua-wd) [reply Gemfile.tt:62]: That's right @andyduong1920, `simplecov-json` is a formatter (the `JSONFormatter` in `simplecov.rb`) :
```ruby
SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter.new([SimpleCov::Formatter::HTMLFormatter, SimpleCov::Formatter::JSONFormatter
```

File: spec/support/simplecov.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 require 'simplecov'
-SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter.new([SimpleCov::Formatter::HTMLFormatter])
+require 'simplecov-json'
+SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter.new([SimpleCov::Formatter::HTMLFormatter, SimpleCov::Formatter::JSONFormatter])
```
Comment (hoangmirs) [inline spec/support/simplecov.rb:3]: Should we split them into 2 lines?
```suggestion
formatters = [SimpleCov::Formatter::HTMLFormatter, SimpleCov::Formatter::JSONFormatter]

SimpleCov.formatter = SimpleCov::Formatter::MultiFormatter.new(formatters)
```

Comment (neua-wd) [reply spec/support/simplecov.rb]: 👍   @hoangmirs. Done in [4fc2d0d](https://github.com/nimblehq/rails-templates/pull/229/commits/4fc2d0dd6a5ba2eba730af18ae6047c0e3008a48)

**Review by andyduong1920 — APPROVED** [review]

**Review by longnd — APPROVED** [review]

**Pull request merged**